worker/uniter/runner/context/env_test.go

```diff
@@ -110,24 +110,25 @@ func (s *EnvSuite) getContext(newProxyOnly bool) (ctx *context.HookContext, expe
 	// It doesn't make sense that we set both legacy and juju proxy
 	// settings, but by setting both to different values, we can see
 	// what the environment values are.
-	return context.NewModelHookContext(
-		"some-context-id",
-		"some-hook-name",
-		"model-uuid-deadbeef",
-		"some-model-name",
-		"this-unit/123",
-		"PURPLE",
-		"proceed with care",
-		"essential",
-		"some-zone",
-		[]string{"he.re:12345", "the.re:23456"},
-		legacyProxy, jujuProxy,
-		names.NewMachineTag("42"),
-	), expected
+	return context.NewModelHookContext(context.ModelHookContextParams{
+		ID:                  "some-context-id",
+		HookName:            "some-hook-name",
+		ModelUUID:           "model-uuid-deadbeef",
+		ModelName:           "some-model-name",
+		UnitName:            "this-unit/123",
+		MeterCode:           "PURPLE",
+		MeterInfo:           "proceed with care",
+		SLALevel:            "essential",
+		AvailZone:           "some-zone",
+		APIAddresses:        []string{"he.re:12345", "the.re:23456"},
+		LegacyProxySettings: legacyProxy,
+		JujuProxySettings:   jujuProxy,
+		MachineTag:          names.NewMachineTag("42"),
+	}), expected
```

Comment:
This is just putting the names here, right? (i find it *way* easier to understand with the fields)' but I can't tell if you added anything to it.

Reply:
This is a purely cosmetic drive-by fix.  All it does is to replace an insanely long arg list expected by `NewModelHookContext` with a struct just in case we want to add more fields in the future.

Reply:
great, that's what it looked like, and I really appreciate the change.